transaction_integration_test.go

```diff
@@ -250,6 +250,150 @@ func TestTransactionCreateWhenGatewayRejectedFraud(t *testing.T) {
 	}
 }
 
+func TestTransactionCreatedWhenAVSBankDoesNotSupport(t *testing.T) {
+	t.Parallel()
+
+	_, err := testGateway.Transaction().Create(&TransactionRequest{
+		MerchantAccountId: avsAndCVVTestMerchantAccountId,
+		Type:              "sale",
+		Amount:            randomAmount(),
+		CreditCard: &CreditCard{
+			Number:         testCreditCards["visa"].Number,
+			ExpirationDate: "05/14",
+			CVV:            "100",
+		},
+
+		BillingAddress: &Address{
+			StreetAddress: "1 E Main St",
+			Locality:      "Chicago",
+			Region:        "IL",
+			PostalCode:    "30001", // Should cause AVS bank does not support error response.
+		},
+	})
+
+	if err == nil {
+		t.Fatal("Did not receive error when creating invalid transaction")
+	}
+
+	if err.Error() != "Gateway Rejected: avs" {
+		t.Fatal(err)
+	}
+
+	txnID := err.(*BraintreeError).Transaction.Id
+	txn, err := testGateway.Transaction().Find(txnID)
```

Comment:
Agreed — I based it on the code added in this commit. I'll update that code as well https://github.com/Figure53/braintree-go/commit/300002fbbb52629b1c09d313b9996bd01639a00b#diff-1ec59924dc3de25f061095c8a4eefdc1R233.